### PR TITLE
[Castle Nathria] Soul infusion icon fix and channeld in Timeline

### PR DESCRIPTION
**Pull request by philipsabri**
Before:
image

Now:
image

Adding the spell to SPELLS did not fix the icon bug, there is a whole discussion about it in discord, but this is how it looks when I comment it out from BAD_ICONS to prove it.
image

### CODE REVIEW

**Comment by philipsabri** [comment]
I had to update the test with these commands, if I did something wrong I will fix it asap
`yarn test`
`u`

File: src/parser/shared/modules/spells/SoulInfusion.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import SPELLS from 'common/SPELLS';
+import CoreChanneling from 'parser/shared/modules/Channeling';
+import Events, {RemoveBuffEvent, CastEvent} from 'parser/core/Events';
+import { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+
+class SoulInfusion extends CoreChanneling {
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER), this.onRemoveBuff);
```
Comment (Khazakdk) [inline src/parser/shared/modules/spells/SoulInfusion.tsx]:
```suggestion
    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.SOUL_INFUSION), this.onRemoveBuff);
```

You can do this to filter the events sent to the listener and remove the need to check the spell id first thing in the listener

**Review by Khazakdk — DISMISSED** [review]
The snapshot changes here are concerning.  The downtime statistic changes drastically for several specs, which shouldn't be happening unless they all happen to use SKS for their snapshots but that shouldn't be the case given our disclaimer on SKS.

File: src/parser/shared/modules/spells/SoulInfusion.tsx (hunk 1 above)
Comment (Sharrq) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: pretty sure i used Hungering/Shriek/etc for the mage ones. and it seems like its removing an entire section from the mage ones?

Comment (Sharrq) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: yea i used Shriek for all 3 mage specs. so those snapshots should not be getting changed at all

Comment (philipsabri) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: Rewrote my bad code @Khazakdk 

I have no idea about the tests tho, reverted them incase I should add or change something @Sharrq

Comment (Sharrq) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: So you have to do the tests, cause the PR wont pass without them

The problem is something in your changes is causing other spec's snapshots to change, and it probably shouldnt.

**Comment by Sharrq** [comment]
Much better.

FYI @Pewtro your hunter tests are getting updated in here cause they are based on Sun King for Multi Target. not sure if you wanted to check to make sure its ok.

The rest of it looks fine to me

File: src/parser/shared/modules/spells/SoulInfusion.tsx (hunk 1 above)
Comment (philipsabri) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: Should be better now!

File: src/parser/shared/modules/spells/SoulInfusion.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+/* Soul Infusion for Sun king's Salvation */
+/* Example Data: https://Wowanalyzer.com/report/g4Pja6pLHnmQtbvk/32-Normal+Sun+King's+Salvation+-+Kill+(10:14)/Pjurbo/standard/timeline */
+
+class SoulInfusion extends CoreChanneling {
+
+  constructor(options: Options) {
+    super(options);
+    
+    const boss = this.owner.boss;
+    this.active = (boss === bosses.bosses["SunKingsSalvation"]) ? true : false;
```
Comment (philipsabri) [inline src/parser/shared/modules/spells/SoulInfusion.tsx]: Could this be done better?

Comment (Sharrq) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: Im sure we could add a BOSSES file to src/game/shadowlands and make that simpler. I cant think of a way to make it any simpler with what is already available.

i dont think there are many, if any, instances where we have analysis for a specific encounter so i dont think we ever made anything for it.

Comment (Sharrq) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: actually you could probably do something like

`this.active = getBossName(this.owner.fight, false) === "Sun King's Salvation"`

not sure that that is any cleaner tho

Comment (philipsabri) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: > Im sure we could add a BOSSES file

> i dont think there are many, if any, instances where we have analysis for a specific encounter so i dont think we ever made anything for it.

Yeah, BOSSES file would make this look the cleanest, but if its not going to get used maybe its not worth to implement it.

Comment (Pewtro) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: Is there a reason to create a ternary for this? Isn't it enough to simply do ``this.active = (boss === bosses.bosses["SunKingsSalvation"])``?

**Comment by Pewtro** [comment]
> Much better.
> 
> FYI @Pewtro your hunter tests are getting updated in here cause they are based on Sun King for Multi Target. not sure if you wanted to check to make sure its ok.
> 
> The rest of it looks fine to me

Yeah seems fine to me

File: src/parser/shared/modules/spells/SoulInfusion.tsx (hunk 2 above)
Comment (abelito75) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: Does `this.active = (boss === bosses.bosses["SunKingsSalvation"]` === `(boss === bosses.bosses["SunKingsSalvation"]) ? true : false;`? Meaning there is no need to do the whole ? : aspect

Comment (philipsabri) [reply src/parser/shared/modules/spells/SoulInfusion.tsx]: Fixed. Overthinked it for sure.

**Comment by Pewtro** [comment]
Thanks!

**Pull request merged**